pyani_plus/methods/method_anim.py

```diff
@@ -0,0 +1,271 @@
+    alignment 2 identity weighted = (6 + 6) - (2 * 0) = 12
+
+    - Calculate %ID
+    (11 + 12) / (10 + 11 + 6 + 6) = 0.696969696969697
+
+    To calculate alignment lengths, we extract the regions of each alignment
+    (either for query or reference) provided in the .delta file and merge the overlapping
+    regions with IntervalTree. Then, we calculate the total sum of all aligned regions.
+    """
+    sim_error = 0  # Hold a count for similarity errors
+
+    regions_ref = defaultdict(list)  # Hold a dictionary for query regions
+    regions_qry = defaultdict(list)  # Hold a dictionary for query regions
+
+    aligned_bases = 0  # Hold a count of aligned bases for each sequence
+    weighted_identical_bases = 0  # Hold a count of weighted identical bases
+
+    for line in [_.strip().split() for _ in filename.open("r").readlines()]:
+        if line[0] == "NUCMER":  # Skip headers
+            continue
+        # Lines starting with ">" indicate which sequences are aligned
+        if line[0].startswith(">"):
+            current_ref = line[0].strip(">")
+            current_qry = line[1]
+        # Lines with seven columns are alignment region headers:
+        if len(line) == 7:  # noqa: PLR2004
+            # Obtaining aligned regions needed to check for overlaps
+            regions_ref[current_ref].append(
+                tuple(sorted([int(line[0]), int(line[1])]))
+            )  # aligned regions reference
+            regions_qry[current_qry].append(
+                tuple(sorted([int(line[2]), int(line[3])]))
+            )  # aligned regions qry
+
+            # Calculate aligned bases for each sequence
+            ref_aln_lengths = abs(int(line[1]) - int(line[0])) + 1
+            qry_aln_lengths = abs(int(line[3]) - int(line[2])) + 1
+            aligned_bases += ref_aln_lengths + qry_aln_lengths
+
+            # Calculate weighted identical bases
+            sim_error += int(line[4])
+            weighted_identical_bases += (ref_aln_lengths + qry_aln_lengths) - (
+                2 * int(line[5])
+            )
+
+    # Calculate average %ID
+    avrg_identity = weighted_identical_bases / aligned_bases
+
+    return (
+        get_aligned_bases_count(regions_ref),
+        get_aligned_bases_count(regions_qry),
+        avrg_identity,
+        sim_error,
+    )
+
+
+def collect_results_directory(deltafilter: Path, indir: Path) -> ComparisonResult:
+    """Return a ComparisonResult for a completed NUCmer comparison.
+
+    The passed directory should contain the output of nucmer comparisons for a single
+    run on a set of input sequences. Each comparison is stored in a separate file,
+    parsed to obtain the count of aligned bases from the query and reference
+    sequences, similarity errors, and average nucleotide identity for each comparison.
+
+    :param deltafilter: Path to the filter file
+    :param indir: Path to input directory (FASTA files)
+    """
+    genome_lengths = {
```

Comment:
This might not be needed if we decide to use `md5` hashes in the file names of comparisons, and not filenames (eg. `<genome_1_md5>_vs_<genome_2_md5>.filter` rather `<genome_1_vs_genome_2>.filter`).